### PR TITLE
make sure by isa Function or a vector of functions

### PR DESCRIPTION
**Pull request by bkamins**
Fixes https://github.com/JuliaData/DataFrames.jl/issues/2500

### CODE REVIEW

File: src/abstractdataframe/sort.jl [hunk 1]
```diff
@@ -473,7 +473,7 @@ julia> sortperm(df, (:x, :y), rev=true)
 """
 function Base.sortperm(df::AbstractDataFrame, cols=[];
                   alg=nothing, lt=isless, by=identity, rev=false, order=Forward)
-    if !(isa(by, Function) || eltype(by) <: Function)
+    if !(isa(by, Function) || (by isa AbstractVector && eltype(by) <: Function))
```
Comment (pdeffebach) [inline src/abstractdataframe/sort.jl:476]: Might as well make it `<: Callable`?

Comment (bkamins) [reply src/abstractdataframe/sort.jl:476]: OK

Comment (bkamins) [inline src/abstractdataframe/sort.jl:476]:
```suggestion
    if !(by isa Base.Callable || (by isa AbstractVector && eltype(by) <: Base.Callable))
```

**Comment by bkamins** [comment]
@pdeffebach - you can approve if you are OK with the changes. Thank you!

**Comment by bkamins** [comment]
Thank you!

**Pull request merged**